devito/types/petsc.py

```diff
@@ -80,6 +89,18 @@ def _C_ctype(self):
     def _C_name(self):
         return self.name
 
+    @property
```

Comment:
These three should not be necessary; they should be inherited from the superclass. If not, they should be added in there

Reply:
I added them to `ArrayBasic`, and removed from `Array`